### PR TITLE
Concordance: tokenization and small fixes

### LINKED ISSUE #208: Concordance: run internal tokenization
##### Text version

0.2.4-dev

##### Orange version

3.5.dev

##### Expected behavior
Concordance ignores corpus.tokens and builds the view independently.

##### Actual behavior
Concordance takes corpus.tokens, which after Preprocessing breaks the view.

##### Steps to reproduce the behavior
Corpus - Preprocess Text (normalize) - Concordance

##### Additional info (worksheets, data, screenshots, ...)
Add also 'startswith' checkbox. Disable exact matches.

### PR DESCRIPTION
**Pull request by ajdapretnar**
##### Issue
Fixes #208. 

##### Description of changes
1. Concordance computes tokens internally to distinguish from corpus.tokens.
2. Input and output set to Corpus.
3. Widget doesn't crash upon deleting the connection.

##### Includes
- [X] Code changes
- [x] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by ajdapretnar** [comment]
@janezd Crucial bug: when deleting a connection from, say, Corpus to Concordance (with a selection in Concordance), the widget crashes. Problem in modelReset.emit(). How to go about fixing this?

File: orangecontrib/text/widgets/owconcordance.py [hunk 1]
```diff
@@ -260,16 +264,14 @@ def update_widget(self):
...
-            self.n_documents = len(self.corpus)
             self.n_matching = '{}/{}'.format(
                 self.model.matching_docs() if self.word else 0,
-                self.n_documents)
-            self.n_tokens = sum(map(len, self.corpus.tokens)) \
-                if self.corpus.has_tokens() else 'n/a'
+                len(self.corpus))
+            self.n_tokens = sum(map(len, self.model.tokens)) \
+                if self.model.tokens is not None else 'n/a'
             self.n_types = len(self.corpus.dictionary) \
```
Comment (nikicc) [inline orangecontrib/text/widgets/owconcordance.py]: We should probably add something like `self.n_types = len(set(tokens))` to ConcordanceModel's set_tokens method and use that instead.

**Comment by ajdapretnar** [comment]
Yaaay, after rebasing I have inadvertedly lost @janezd tests. 😞 How do I fix this and why on Earth does Git override someone else's commit??? That's bs. 😠

**Comment by ajdapretnar** [comment]
@nikicc After a lot of painful edits, this is ready to merge. :)

**Pull request merged**